### PR TITLE
docs: Add documentation for S3_DIRS Environment Variable

### LINKED ISSUE #317: Add S3 code mounting user guide
## Suggestion

We currently only mention this feature in configuration.md, for the config variable. A feature which is as useful as this, should be properly documented, and have a user guide.

## Example

This docker-compose example shows how to set this variable, the rest is potentially left to the writer of the user guide:
```
version: "3.8"

services:
  localstack:
    container_name: "${LOCALSTACK_DOCKER_NAME-localstack_main}"
    image: localstack/localstack
    ports:
      - "127.0.0.1:4566:4566"            # LocalStack Gateway
      - "127.0.0.1:4510-4559:4510-4559"  # external services port range
      - "127.0.0.1:53:53"                # DNS config (only required for Pro)
      - "127.0.0.1:53:53/udp"            # DNS config (only required for Pro)
      - "127.0.0.1:443:443"              # LocalStack HTTPS Gateway (only required for Pro)
    environment:
      - DEBUG=${DEBUG-}
      - PERSISTENCE=${PERSISTENCE-}
      - LAMBDA_EXECUTOR=${LAMBDA_EXECUTOR-}
      - LOCALSTACK_API_KEY=${LOCALSTACK_API_KEY-}  # only required for Pro
      - DOCKER_HOST=unix:///var/run/docker.sock
      - S3_DIR=/tmp/s3-mounts
    volumes:
      - "${LOCALSTACK_VOLUME_DIR:-./volume}:/var/lib/localstack"
      - "./mounts:/tmp/s3-mounts"
      - "/var/run/docker.sock:/var/run/docker.sock"
```

### PR DESCRIPTION
**Pull request by snhardin**
Fixes #317 

Adds a section in the S3 service documentation about proper usage of the `S3_DIRS` configuration variable with multiple examples.

### CODE REVIEW

File: content/en/localstack/configuration.md [hunk 1]
```diff
@@ -181,7 +181,7 @@ While the ElasticSearch API is actively maintained, the configuration variables
 
 | Variable | Example Values | Description |
 | - | - | - |
-| `S3_DIR` | | Configure a global parent directory that contains all buckets as sub-directories (`S3_DIR=/path/to/root`) or an individual directory that will get mounted as special bucket names (`S3_DIR=/path/to/root/bucket1:bucket1`) |
+| `S3_DIR` | | Configure a global parent directory that contains all buckets as sub-directories (`S3_DIR=/path/to/root`) or an individual directory that will get mounted as special bucket names (`S3_DIR=/path/to/root/bucket1:bucket1`). Only available for Localstack Pro. |
```
Comment (snhardin) [inline content/en/localstack/configuration.md:184]: Hi Localstack Team. Please clarify if this feature is for Pro only or not. I was having trouble getting this working in the free tier and I think the original PR adding this documentation mentions that this is a Pro-only feature. Admittedly, I was trying multiple things when I added my Pro key to the configuration that I'm not sure if I simply had my Localstack instance misconfigured or if adding the key caused this to start working.

File: content/en/aws/s3/index.md [hunk 2]
```diff
@@ -58,3 +58,81 @@ As a special case in LocalStack, leaving out `.s3.<region>` also works for the `
...
+      - PERSISTENCE=${PERSISTENCE-}
+      - LAMBDA_EXECUTOR=${LAMBDA_EXECUTOR-}
+      - LOCALSTACK_API_KEY=${LOCALSTACK_API_KEY-}  # only required for Pro
+      - DOCKER_HOST=unix:///var/run/docker.sock
+      - S3_DIR=/tmp/buckets/test-bucket:my-test-bucket,/tmp/buckets/other-bucket:my-other-bucket
+    volumes:
+      - "${LOCALSTACK_VOLUME_DIR:-./volume}:/var/lib/localstack"
+      - "./buckets:/tmp/buckets"
+      - "/var/run/docker.sock:/var/run/docker.sock"
+```
```
Comment (HarshCasper) [inline content/en/aws/s3/index.md:138]: I would prefer to have these examples removed out. In the previous section, we have given enough context to the users to get started up, and these examples just repeat the information.

File: content/en/localstack/configuration.md (hunk 1 above)
Comment (HarshCasper) [reply content/en/localstack/configuration.md:184]: @thrau Do you think mentioning a fleeting note if a config is only for Pro, would be fine?

File: content/en/aws/s3/index.md [hunk 3]
```diff
@@ -58,3 +58,81 @@ As a special case in LocalStack, leaving out `.s3.<region>` also works for the `
...
 All other requests will be considered path-style requests.
 {{% /alert %}}
+
+## Storage Configuration
+
+Localstack Pro can be configured to store S3 files in specific locations on the filesystem. This can be useful in many different ways, including moving multiple files into place in one or more S3 buckets without needing to use `awslocal` to upload them. Localstack will use the path(s) specified in the `S3_DIRS` environment variable to store files within the container. Paired with a docker mount, this can be used to store files in the path of choice on the filesystem.
+
+Paths may be specified in two different configurations. Specifying only a path will store all S3 buckets in that path:
+
+```
```
Comment (HarshCasper) [inline content/en/aws/s3/index.md:68]: Specify the syntax highlighting: `sh` / `bash` in this case

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: content/en/aws/s3/index.md (hunk 3 above)
Comment (snhardin) [reply content/en/aws/s3/index.md]: Fixed in latest commit. 👍

File: content/en/aws/s3/index.md (hunk 2 above)
Comment (snhardin) [reply content/en/aws/s3/index.md]: Fixed in latest commit. 👍

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**